open_prices/proofs/utils.py

```diff
@@ -124,3 +133,73 @@ def store_file(
     # Build file_path
     file_path = generate_relative_path(current_dir_id_str, file_stem, extension)
     return (file_path, mimetype, image_thumb_path)
+
+
+def run_ocr_on_image(image_path: Path | str, api_key: str) -> dict[str, Any] | None:
+    """Run Google Cloud Vision OCR on the image stored at the given path.
+
+    :param image_path: the path to the image
+    :param api_key: the Google Cloud Vision API key
+    :return: the OCR data as a dict or None if an error occurred
+    """
+    with open(image_path, "rb") as f:
+        image_bytes = f.read()
+
+    base64_content = base64.b64encode(image_bytes).decode("utf-8")
+    url = f"https://vision.googleapis.com/v1/images:annotate?key={api_key}"
+    r = http_session.post(
+        url,
+        json={
+            "requests": [
+                {
+                    "features": [{"type": "TEXT_DETECTION"}],
+                    "image": {"content": base64_content},
+                }
+            ]
+        },
+    )
+
+    if not r.ok:
+        logger.debug(
+            "Error running OCR on image %s, HTTP %s\n%s",
+            image_path,
+            r.status_code,
+            r.text,
+        )
+    return r.json()
+
+
+def run_ocr_task(image_path: Path | str, override: bool = False) -> None:
+    """Run OCR on the image stored at the given path and save the result to a
+    JSON file.
+
+    The JSON file will be saved in the same directory as the image, with the
+    same name but a `.json` extension.
+
+    :param image_path: the path to the image
+    :param override: whether to override existing OCR data, default to False
+    """
+    image_path = Path(image_path)
+    api_key = settings.GOOGLE_CLOUD_VISION_API_KEY
+
+    if api_key is None:
+        logger.error("No Google Cloud Vision API key found")
+        return
+
+    ocr_json_path = image_path.with_suffix(".json.gz")
+
+    if ocr_json_path.exists() and not override:
+        logger.info("OCR data already exists for %s", image_path)
+        return
+
+    data = run_ocr_on_image(image_path, api_key)
+
+    if data is None:
+        return
+
+    data["created_at"] = int(time.time())
+
+    with gzip.open(ocr_json_path, "wt") as f:
+        f.write(json.dumps(data))
```

Comment:
so it stores the result in a jsonl.gz file next to the image ?

Reply:
if we re-run on the same image, will it override ?

Reply:
> so it stores the result in a jsonl.gz file next to the image ?
Yes exactly!

Reply:
> if we re-run on the same image, will it override ?

It depends on the value of `override`

Reply:
ah ouiii bien vu !